src/cdk/overlay/overlay-directives.ts

```diff
@@ -230,6 +231,15 @@ export class CdkConnectedOverlay implements OnDestroy, OnChanges {
     this._push = coerceBooleanProperty(value);
   }
 
+  /** Whether the overlay should be disposed of when the user goes backwards/forwards in history. */
+  @Input('cdkConnectedOverlayDisposeOnNavigation')
```

Comment:
We're working towards using input transforms for this so let's start doing it here:

```
@Input({alias: 'cdkConnectedOverlayDisposeOnNavigation', transform: booleanAttribute})
```

Reply:
I have only added transform here for `cdkConnectedOverlayDisposeOnNavigation` cause this PR is strictly for mentioned issue, should I send in other PRs with this Input transform refactoring? :grin:

Reply:
Changing it in existing places will require a lot of debugging internally so it'll be easier if we handle it within the team.